### ISSUE TITLE
Is it possible to tag objects in potree?

### ISSUE THREAD

**Issue by straz** [potree/potree#243]
Thank you very much for making Potree available. I have just begun exploring visualization tools, and I hope to be able to use potree.

Our project concerns visualization of 3D neuroanatomical data. We have a point cloud data set representing neurons in the brain. A point cloud may represent millions of neurons, and each neuron has a unique `NEURON_ID`.

I would like to be able to tag points in the cloud to control aspects of visibility, opacity, and color through interactive controls. For example, it would be nice to click on one or more neurons of interest and turn them green and opaque, while all the other neurons remain gray and translucent.

I can see there is a `CLASSIFICATION` attribute in the [loader](https://github.com/potree/potree/blob/master/src/loader/PointAttributes.js), but is there any documentation on how to use it? Can I use this to tag points? How do I add this attribute to data files? Can I use `uint64` values instead of `uint8` values?

I’ve read the documentation in [github](https://github.com/potree/potree/blob/master/docs) but I would welcome some help finding ways to customize the tool. 

1. Is it possible to associate tags with points in potree?
1. Is it possible to add new attributes (like `NEURON_ID`) to tag each point in BINARY file format? 
1. Is it possible to add new attributes to LAS, LAZ, or XYZRGB files?
1. Is it possible to use these new attributes with PotreeConverter? 
1. Is it possible using javascript to select a set of tagged points, and toggle them (visible/invisible)? Or change their RGBA color or opacity?
1. Is it possible to invoke a javascript function with the `NEURON_ID` of a point when the user clicks on the cloud?

many thanks, 
Steve

Steve Strassmann, PhD
Visiting Scientist
Harvard Medical School
617 487 4101
steve_strassmann@hms.harvard.edu

**Comment by straz** [comment]
I'm adding a small (25,000 points) sample pointcloud showing what I'm trying to do. Thanks for any suggestions:
[dots.tar.gz](https://github.com/potree/potree/files/470694/dots.tar.gz)

**Comment by JWGmeligMeyling** [comment]
> Is it possible to associate tags with points in potree?

Yes, kind of. Currently the LAS writer of the PotreeConverter has the ability to send `classification` and `pointSourceId`. The `classification` is usually used in LiDaR to *tag* an object as being a building, or ground. In the [`development`](https://github.com/potree/potree/pull/179/commits/08d7347d770c2861ccd6679ee26a049a83d42e7d) branch work is started to be able to hide points by their classification (issue #157).

> Is it possible to add new attributes (like NEURON_ID) to tag each point in BINARY file format?

The value for `classification` is limited to an `unsigned char` and its values actually have [reserved meanings](http://www.asprs.org/wp-content/uploads/2010/12/asprs_las_format_v12.pdf), so it may not be the most logical attribute to store your value in.

Your best bet would probably be to extend the [`BINPointWriter`](https://github.com/potree/PotreeConverter/blob/master/PotreeConverter/include/BINPointWriter.hpp) so that it sends the attributes that you need to the client. This requires companion changes to the [`BinaryLoader`](https://github.com/potree/potree/blob/master/src/loader/BinaryLoader.js#L69-L91) in the viewer logic as well.

> Is it possible to add new attributes to LAS, LAZ, or XYZRGB files?

The LAS and LAZ formats do not have any free room to play with new attributes, so you can only use these formats if it suffices to reuse existing fields. For XYZRGB I am not sure.

> Is it possible to use these new attributes with PotreeConverter?

Yes, as mentioned above, the writers in the PotreeConverter can be extended to include new attributes.

> Is it possible using javascript to select a set of tagged points, and toggle them (visible/invisible)? Or change their RGBA color or opacity?

Yes, as shown with the classification feature, it is possible to change the RGBA color for the points based on attributes. For classification something is under development, for newly added properties, something custom made is needed. (But the work on the classification feature could be a good starting point).

> Is it possible to invoke a javascript function with the NEURON_ID of a point when the user clicks on the cloud?

This is possible with [`pointcloud.pick`](https://github.com/potree/potree/blob/master/src/utils/MeasuringTool.js#L533). This gives you access to the classification, color, intensity, normal and all other known point attributes.

**Comment by iHunt101** [comment]
Can we tag using point with database?
